fsharp-backend/src/LibExecution/StdLib/LibString.fs

```diff
@@ -406,16 +406,27 @@ let fns : List<BuiltInFn> =
 //      ; sqlSpec = NotYetImplementedTODO
 //      ; previewable = Pure
 //      ; deprecated = NotDeprecated }
-//      { name = fn "String" "reverse" 0
-//      ; parameters = [Param.make "string" TStr]
-//      ; returnType = TStr
-//      ; description = "Reverses `string`"
-//      ; fn =
-//         (function
-//         | _, [DStr s] -> DStr (Unicode_string.rev s) | args -> incorrectArgs ())
-//      ; sqlSpec = NotYetImplementedTODO
-//      ; previewable = Pure
-//      ; deprecated = NotDeprecated }
+    { name = fn "String" "reverse" 0
```

Comment:
I don't think this works right - in the OCaml version we used a unicode reversing function. Here you use a character reversing function. I think this will break for unicode and similar. Can you add tests for it?

I'm sure there's like a String.reverse() method which does this right.

Reply:
There is a rev(string) function, I just tried it, but it does not work properly with Unicode. After reading some documentation, I have implemented ParseCombiningCharacters method to get each character or surrogate index and then work with them. Let me know what do you think :)